packages/hardhat-core/src/internal/core/runtime-environment.ts

```diff
@@ -334,9 +354,14 @@ export class Environment implements HardhatRuntimeEnvironment {
    */
   private _resolveValidTaskArguments(
     taskDefinition: TaskDefinition,
-    taskArguments: TaskArguments
+    taskArguments: TaskArguments,
+    subtaskArguments: SubtaskArguments = {}
```

Comment:
This is a private function so the default value is not necessary.

Reply:
Agreed. It's not undefined when it's called.